backend/BCAPIServer/src/main/java/routes/bookclubs/ListFeed.java

```diff
@@ -1,34 +1,69 @@
 package routes.bookclubs;
 
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
 import daos.Books;
+import daos.User;
+import java.sql.Connection;
 import spark.Request;
 import spark.Response;
 import spark.Route;
 import types.BookPost;
 
 import java.util.List;
+import utils.BCGsonUtils;
 
 public class ListFeed implements Route {
-    private Books books;
 
-    public ListFeed(Books books) {
-        this.books = books;
+  private FirebaseApp fbApp;
+  private Connection sqlConn;
+
+  public ListFeed(FirebaseApp fbApp, Connection sqlConn) {
+    this.fbApp = fbApp;
+    this.sqlConn = sqlConn;
+  }
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    JsonObject respJson = new JsonObject();
+
+    String token = request.queryParams("token");
+    if (token == null) {
+      respJson.addProperty("status", "failure");
+      respJson.addProperty("failure_reason", "Expected fields are missing!");
+      return respJson.toString() + "\n";
     }
 
-    @Override
-    public Object handle(Request request, Response response) throws Exception {
-        JsonObject jsonObject = new JsonObject();
+    FirebaseToken decodedToken;
 
-        String token = request.queryParams("token");
-        // TODO: resolve the indivdual account's posts
+    try {
+      decodedToken = FirebaseAuth.getInstance(fbApp)
+          .verifyIdToken(token, true);
+    } catch (FirebaseAuthException e) {
+      respJson.addProperty("status", "failure");
+      respJson.addProperty("failure_reason", String.valueOf(e.getAuthErrorCode()));
+      return respJson.toString() + "\n";
+    }
+
+    String uid = decodedToken.getUid();
+    User users = new User(uid, sqlConn);
+    Books books = new Books(sqlConn);
 
-        // For now just return a list of all the posts.
-        List<BookPost> posts = books.listBookPosts("*");
-        for (BookPost post : posts) {
-            System.out.println(post.toString());
-        }
+    List<String> subscribedBookKeys = users.getSubscribedClubs();
+    JsonArray postsArr = new JsonArray();
 
-        return jsonObject;
+    for (String bookKey : subscribedBookKeys) {
```

Comment:
are we still adding posts from people you follow? or just book clubs subscribed?

Reply:
I can add that in rq